staff_features/resources/resource_create.feature

```diff
@@ -1,27 +1,41 @@
 Feature: Resource Create
-
   Background:
     Given an administrator user is logged in
-
   Scenario: Resource is created
     Given the user is on the New Resource page
      When the user fills in 'Title'
       And the user fills in 'Identifier'
-      And the user fills in 'Language'
-      And the user fills in 'Expression'
-      And the user fills in 'Level of Description'
-      And the user fills in 'Language of Description'
+      And the user selects 'Class' from 'Level of Description'
+      And the user fills in 'Language' with 'English' and selects 'English' in the 'Languages' form
+      And the user fills in 'Script' with 'Old Hungarian' and selects 'Old Hungarian (Hungarian Runic)' in the 'Languages' form
+      And the user selects 'Single' from 'Type' in the 'Dates' form
+      And the user fills in 'Begin' with '2000-01-01' in the 'Dates' form
+      And the user fills in 'Number' with '123456789' in the 'Extents' form
+      And the user selects 'Cassettes' from 'Type' in the 'Extents' form
+      And the user fills in 'Language of Description' with 'English' and selects 'English' in the 'Finding Aid Data' form
+      And the user fills in 'Script of Description' with 'Old Hungarian' and selects 'Old Hungarian (Hungarian Runic)' in the 'Finding Aid Data' form
```

Comment:
I think we will increase the reusability and readability of the When steps by using the following

    When the user fills in the following fields:
      | field                        | value                             |
      | Title                        | Sample Resource                  |
      | Identifier                   | Unique123                        |
      | Language                     | English                          |
      | Script                       | Old Hungarian (Hungarian Runic)  |
      | Level of Description         | Class                            |
      | Language of Description      | English                          |
      | Script of Description        | Old Hungarian (Hungarian Runic) |
    And the user selects the following options:
      | field               | option    |
      | Type in Dates       | Single    |
      | Type in Extents     | Cassettes |
    And the user fills in the following date and extent fields:
      | field               | value      |
      | Begin               | 2000-01-01|
      | Number in Extents   | 123456789 |
    And the user clicks on 'Save'

Reply:
Unfortunately, I am afraid this is not technically feasible here.

Whether somebody is writing a Gherkin scenario, or trying to read and replicate it, it would be helpful to do that in the order the fields and sub-record forms appear on the screen.
Doing it this way, the user has to fill in a field in one form, then scroll to a different section in the page, then scroll back for a different type of field, which is quite cumbersome and counterintuitive.

At this point, we have the following steps in the system, which work quite well and are perfectly re-usable:

`the user fills in {string} with {string}`
`the user fills in {string} with {string} in the {string} form`
`the user selects {string} from {string}`
`the user selects {string} from {string} in the {string} form`

and on this feature, we are adding:
`the user fills in {string} with {string} and selects {string} in the {string} form`

which extends the DSL of the existing steps, and it's re-usable wherever we have select lists we are generated dynamically according to user input.

Creating something like `the user selects the following options`, and then placing the form inside the field column, makes it impossible to find and distinct the field from the form,
and it also violates re-usability, since the `the user selects {string} from {string} in the {string} form` is already there, and works for every field that exists inside a form s1ection. Doing it this way, we have to create lots of distinct steps with duplicated underlying implementations, that essentially do the same thing.